config/m4/prof.m4

```diff
@@ -0,0 +1,252 @@
+# prof.m4 - Profiling, instrumentation
+# 
+# Copyright (C) Mellanox Technologies Ltd. 2016.  ALL RIGHTS RESERVED.
+# See file LICENSE for terms.
+#
+
+##########################
+# libibprof profiling support
+#
+AC_DEFUN([PROF_IBPROF_SETUP],
+[
+AC_ARG_WITH([ibprof],
+    AC_HELP_STRING([--with-ibprof],
+                   [Search ibprof location (default NO)]),
+    [],
+    [with_ibprof=no]
+)
+
+AS_IF([test "x$with_ibprof" == xno],
+    [],
+    [AC_CHECK_HEADER(
+        [$with_ibprof/include/ibprof_api.h],
+        [
+         CFLAGS="$CFLAGS -DVMA_TIME_IBPROF"
+         CXXFLAGS="$CXXFLAGS -DVMA_TIME_IBPROF"
+         CPPFLAGS="$CPPFLAGS -I$with_ibprof/include"
+         if test -d "$with_ibprof/lib64"; then
+             LDFLAGS="$LDFLAGS -L$with_ibprof/lib64 -Wl,--rpath,$with_ibprof/lib64"
+         else
+             LDFLAGS="$LDFLAGS -L$with_ibprof/lib -Wl,--rpath,$with_ibprof/lib"
+         fi
+         AC_SUBST([LIBIBPROF_LIBS], "-libprof")
+        ],
+        [AC_MSG_ERROR([ibprof support requested, but <$with_ibprof/include/ibprof_api.h> not found.])])
+])
+])
+
+##########################
+#
+# RDTSC measurements support
+#
+# ****** Total VMA RX******** 
+# RDTSC_MEASURE_RX_CQE_RECEIVEFROM
+#
+# ******* Verbs Poll ***********
+# RDTSC_MEASURE_RX_VERBS_IDLE_POLL 
+# RDTSC_MEASURE_RX_VERBS_READY_POLL
+#
+# ******* LWIP ***********
+# RDTSC_MEASURE_RX_LWIP 
+#
+# ******* Other RX ***********
+# RDTSC_MEASURE_RX_DISPATCH_PACKET 
+# RDTSC_MEASURE_RX_AFTER_PROCCESS_BUFFER_TO_RECIVEFROM 
+# RDTSC_MEASURE_RX_VMA_TCP_IDLE_POLL 
+# RDTSC_MEASURE_RX_READY_POLL_TO_LWIP 
+# RDTSC_MEASURE_RX_LWIP_TO_RECEVEFROM 
+#
+# ****** Total VMA TX ******** 
+# RDTSC_MEASURE_TX_SENDTO_TO_AFTER_POST_SEND 
+
+# ******* Verbs Post Send ***********
+# RDTSC_MEASURE_TX_VERBS_POST_SEND 
+
+# ******* App ***********
+# RDTSC_MEASURE_RECEIVEFROM_TO_SENDTO
+#
+AC_DEFUN([PROF_RDTSC_SETUP],
+[
+AC_MSG_CHECKING([if rdtsc-rx-cqe-recvfrom is enabled])
```

Comment:
all of this code is one big duplication
can it be changed to a function that will get the rdtsc-* as a parameter and then just call it few times?